src/RestServer/Controllers/v1/WalletController.cs

```diff
@@ -0,0 +1,646 @@
+// Copyright (C) 2015-2023 The Neo Project.
+//
+// The Neo.Plugins.RestServer is free software distributed under the MIT software license,
+// see the accompanying file LICENSE in the main directory of the
+// project or http://www.opensource.org/licenses/mit-license.php
+// for more details.
+//
+// Redistribution and use in source and binary forms with or without
+// modifications are permitted.
+
+using Akka.Actor;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Neo.Network.P2P.Payloads;
+using Neo.Plugins.RestServer.Exceptions;
+using Neo.Plugins.RestServer.Helpers;
+using Neo.Plugins.RestServer.Models.Error;
+using Neo.Plugins.RestServer.Models.Wallet;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using Neo.Wallets;
+using Neo.Wallets.NEP6;
+using System.Net.Mime;
+using System.Numerics;
+
+namespace Neo.Plugins.RestServer.Controllers.v1
+{
+    /// <summary>
+    /// Wallet API
+    /// </summary>
+    [Route("/api/v{version:apiVersion}/wallet")]
+    [DisableCors]
+    [Produces(MediaTypeNames.Application.Json)]
+    [Consumes(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(ErrorModel))]
+    [ApiVersion("1.0")]
+    [ApiController]
+    public class WalletController : ControllerBase
+    {
+        internal static WalletSessionManager WalletSessions { get; } = new();
+
+        private readonly NeoSystem _neosystem;
+
+        /// <summary>
+        /// CTOR
+        /// </summary>
+        /// <exception cref="NodeNetworkException">Node network doesn't match plugins network.</exception>
+        public WalletController()
+        {
+            _neosystem = RestServerPlugin.NeoSystem ?? throw new NodeNetworkException();
+        }
+
+        /// <summary>
+        /// Opens a wallet.
+        /// </summary>
+        /// <param name="model"></param>
+        /// <returns>A newly created wallet session object.</returns>
+        /// <response code="200">Returns newly create wallet session object.</response>
+        /// <response code="400">An error occurred. See Response for details.</response>
+        [HttpPost("open", Name = "WalletOpen")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(WalletSessionModel))]
+        public IActionResult WalletOpen(
+            [FromBody]
+            WalletOpenModel model)
+        {
+            if (new FileInfo(model.Path).DirectoryName.StartsWith(AppContext.BaseDirectory, StringComparison.InvariantCultureIgnoreCase) == false)
+                throw new UnauthorizedAccessException(model.Path);
```

Comment:
Why the wallet has to be in subdirectory of CLI root path? It can be anywhere.

Reply:
https://github.com/neo-project/neo-modules/pull/810#discussion_r1301965681

Reply:
I prefer it to be remove, but as you see in comment. Others want it removed and so does `code` analysis.

Reply:
OK, I see. Let's keep it.

Reply:
Can we at least make the path range in CLI's root path? `AppContext.BaseDirectory` is the subdirectory path.
![1703668908928](https://github.com/neo-project/neo-modules/assets/38493437/3ab6e966-ad33-4881-ba9f-66b8e343e8b0)

Reply:
base path is where ever you run `neo-cli.exe` at.

Reply:
> base path is where ever you run `neo-cli.exe` at.

The `new FileInfo(model.Path).DirectoryName` path doesn't have `\\` in the end, so if I'd like to open a wallet which start with `AppContext.BaseDirectory`, it need be included in a subdirectory of Neo-CLI. Can you see the difference?
![1704248164106](https://github.com/neo-project/neo-modules/assets/38493437/f2cf8f43-29a5-4b1d-b62a-3c5f034cf2e4)
![1704248188652](https://github.com/neo-project/neo-modules/assets/38493437/e699b4fe-c965-4492-a2c7-f6d8aff11a97)
![1704248268920](https://github.com/neo-project/neo-modules/assets/38493437/4f3fdf66-4eb2-4218-803e-03e24f130be1)

Reply:
Yes i can fix that.

Reply:
Fixed Problem. Can you test again?
![image](https://github.com/neo-project/neo-modules/assets/8141309/98dff455-a32e-4bfd-aca6-ba61c2b6828d)